CMakePresets.json

```diff
@@ -29,24 +21,64 @@
     {
       "name": "gcc-debug",
       "displayName": "GCC Debug Build",
+      "inherits": ["_root-config"],
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER": "g++",
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_CXX_FLAGS": "-fsanitize=address -fsanitize=pointer-compare -fsanitize=pointer-subtract -fsanitize=leak -fsanitize=undefined"
+      }
+    },
+    {
+      "name": "gcc-release",
+      "displayName": "GCC Release Build",
       "inherits": [
         "_root-config",
-        "_debug-base"
+        "_release-base"
       ],
       "cacheVariables": {
         "CMAKE_CXX_COMPILER": "g++"
       }
     },
     {
-      "name": "gcc-release",
-      "displayName": "GCC Release Build",
+      "name": "xcode-debug",
```

Comment:
The name is sourced per your suggestion. 
See: https://discourse.bemanproject.org/t/what-do-we-want-our-presets-to-be/246

Reply:
There is a reason to separate apple clang out in respect to clang-clang, it is to prevent scenaior like: https://github.com/bemanproject/exemplar/issues/65 .

The main pain-point is sanitizers.

Reply:
I take that your initial proposal is to support xcode based toolchain. Apple-clang is just part of it.

The intention here is not to soly support apple clang but extra configurations that might arise from xcode based toolchains.

Let me know if I am interpreting your initial proposal incorrectly.
![image](https://github.com/user-attachments/assets/82f728d8-2413-4dfa-8b8a-220f48b41267)

Your proposal is linked above as well.

Reply:
Ah, I see the confusion now. My proposal was suggesting that the *Generator* be "XCode" for the MacOS platform and that the *Generator* be "Visual Studio 2022" for the Windows platform.

What you've done here is use "Ninja" as the generator for these platforms. I think that's fine for now.

Reply:
What should the preset name be then?

AppleClang-debug?

Reply:
> AppleClang-debug?

Yep, that sounds good.

For compiler names we could potentially follow the [convention here](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html). That implies we would change gcc-release to GNU-release which may be surprising for those not familiar with that list, but at least we'd have something to reference.

Reply:
Yeah, I'm with you there.

Reply:
Done

Reply:
hum, should I update `msvc-debug` to `MSVC-debug`, or should I use `appleclang-debug`?